File: Client/models/AttributeTypeUnitDTO.ts
export interface AttributeTypeUnitGetDTO {
  id: number;
  value: string;
}
export interface AttributeTypeUnitPostDTO {
  value: string;
  attributeTypeId: number;
}

export interface AttributeTypeUnitPatchDTO {
  id: number;
  value: string;
}

File: Server/PublicApi/1.0/AttributeTypeUnitDTO.cs
using System.ComponentModel.DataAnnotations;
using DAL.Base;
using DAL.Base.Entities;

namespace PublicApi._1._0
{
    public class AttributeTypeUnitGetDTO : DomainEntityId
    {
        public string Value { get; set; } = default!;
    }

    public class AttributeTypeUnitPostDTO
    {
        [Required] public string Value { get; set; } = default!;
        [Required] public long AttributeTypeId { get; set; } = default!;
    }

    public class AttributeTypeUnitPatchDTO : DomainEntityId
    {
        [Required] public string Value { get; set; } = default!;
    }
}